homeassistant/components/tuya/binary_sensor.py

```diff
@@ -285,6 +285,16 @@ class TuyaBinarySensorEntityDescription(BinarySensorEntityDescription):
         ),
         TAMPER_BINARY_SENSOR,
     ),
+    # Thermostat
+    # https://developer.tuya.com/en/docs/iot/f?id=K9gf45ld5l0t9
+    "wk": (
+        TuyaBinarySensorEntityDescription(
+            key=DPCode.WINDOWS_STATE,
```

Comment:
I cannot find `windows_state` in the linked documentation.

Reply:
Potentially this is missing from the documentation then as it is returned by the API. Consult the API response:
```
  "result": [
    {
      "active_time": 1668073884,
      "biz_type": 18,
      "category": "wk",
      "create_time": 1668073884,
      "icon": "smart/icon/ay1523512777050bLIId/2ccfa5a4faa500e31133a147f1875daa.jpg",
      "id": "bfc16dfe85c89dd018ua8e",
      "ip": "***",
      "lat": "***",
      "local_key": "***",
      "lon": "***",
      "model": "HY36xRT Zigbee",
      "name": "Sypialnia",
      "node_id": "04cd15fffe3b802d",
      "online": true,
      "owner_id": "70540555",
      "product_id": "cpmgn2cf",
      "product_name": "SmartTRV",
      "status": [
        {
          "code": "temp_set",
          "value": 200
        },
        {
          "code": "temp_current",
          "value": 190
        },
        {
          "code": "mode",
          "value": "auto"
        },
        {
          "code": "child_lock",
          "value": false
        },
        {
          "code": "fault",
          "value": 0
        },
        {
          "code": "roomtemp_calibrat",
          "value": 0
        },
        {
          "code": "lowtemp",
          "value": 5
        },
        {
          "code": "hightemp",
          "value": 35
        },
        {
          "code": "wind",
          "value": "AAUK"
        },
        {
          "code": "boost",
          "value": 300
        },
        {
          "code": "valve_set",
          "value": "normal"
        },
        {
          "code": "comfort_temp",
          "value": 20
        },
        {
          "code": "eco_temp",
          "value": 15
        },
        {
          "code": "valve",
          "value": 10
        },
        {
          "code": "power_state",
          "value": false
        },
        {
          "code": "week_state",
          "value": "2"
        },
        {
          "code": "Prog_Workday",
          "value": "BgAUCAAUCx4UDB4UEB4RFAAR"
        },
        {
          "code": "Prog_Restday",
          "value": "BgAUCAAPCx4PDB4PER4UFgAP"
        },
        {
          "code": "Temp_holiday",
          "value": 15
        },
        {
          "code": "windows_state",
          "value": false
        },
        {
          "code": "Auto_Lock",
          "value": false
        },
        {
          "code": "Days_Holiday",
          "value": 1
        }
      ],
      "sub": true,
      "time_zone": "+01:00",
      "uid": "***",
      "update_time": 1668073903,
      "uuid": "04cd15fffe3b802d"
    },

```

Reply:
It could be a customization created by the manufacturer, would be nice if there was a specification. In general, we want to avoid adding these.

Reply:
The device works well with Tuya's `Smart Life` application (and within this application these values are exposed), hence I assume that it is intended behavior. This is quite common model I believe, Moes HY360RT/TS0601. 

I've also checked the Tuya IoT portal and all these parameters are there...

Reply:
@frenck After some extra investigation:
1. The link https://developer.tuya.com/en/docs/iot/f?id=K9gf45ld5l0t9 refers to `Standard Instruction Set` only.
2. I've checked the Tuya IoT platform and noticed that the platform allows selecting `Control Instruction Mode`:
  - **Standard Instruction**: "Control devices with standard instructions after mapping. Various devices across manufacturers can be controlled with a single set of instructions, but note that some functions might not be compatible."
  - **DP Instruction** (it's selected in my case, not sure at which point, maybe it was a default): "Control devices with original DP instructions. You can access any DP information."> 
 
The IoT portal suggests both sets are the same (different from [Standard Instruction Set](https://developer.tuya.com/en/docs/iot/f?id=K9gf45ld5l0t9)):
DP Instruction | Standard Instruction
-- | --
temp_set | temp_set
temp_current | temp_current
mode | mode
child_lock | child_lock
fault | fault
roomtemp_calibrat | roomtemp_calibrat
lowtemp | lowtemp
hightemp | hightemp
wind | wind
boost | boost
valve_set | valve_set
comfort_temp | comfort_temp
eco_temp | eco_temp
valve | valve
power_state | power_state
week_state | week_state
Prog_Workday | Prog_Workday
Prog_Restday | Prog_Restday
Temp_holiday | Temp_holiday
windows_state | windows_state
Auto_Lock | Auto_Lock
Days_Holiday | Days_Holiday

3. I didn't get any reply for the tech support ticket.

Reply:
@frenck any update? 
Note that adding the DP instruction set is rather safe as on the startup, the actually available entities are discovered. I know that there is a performance impact, but this is only for a set of string comparisions. Moreover to fully utilize Tuya Thermostat, it definitely makes sens to enable the DP instruction set.

Reply:
@frenck Is there a chance that this change will be accepted?